HW4: Integration with Team12 repo, opensource_cpp_base

123R3N321 opened the pull request:
Not entirely sure what to do with it next.

123R3N321 committed:
Add opensource_cpp_base submodule.

git submodule add <ssh/url> from Implementation branch.

123R3N321 committed:
Update README.md

Add comments and instructions regarding integration.

kyotov commented:
looks good!

this is enough in terms of what the course needs.
but in real life i would consider a new repo for the app where the two components -- yours and theirs are both submodules.
just FYI

123R3N321 committed:
Update Graph.cpp.

Change undirected graph to bi-directional. Potentially breaking bfs and dfs

123R3N321 committed:
Update README.md

Warn about potentially broken bfs/dfs/tree iteration.

123R3N321 committed:
Update pyproject.toml and README.md

Switch to uv as package manager per prof's request. Nothing really changed.

123R3N321 committed:
Modify Graph.cpp

Use boost implementation of bfs and dfs instead of own implementation.
Note that I did not read  default_visitor implementation. Do not garantee better performance.

123R3N321 committed:
Add integration test.

123R3N321 committed:
Modify sub-module for integration
